### PR TITLE
Use `internal-encryption` to deploy internal certificates automatically

### PR DESCRIPTION
**Pull request by nak3**
This patch is same purpose with https://github.com/knative/serving/pull/13005.

Currently users have to deploy certificates manually with several options such as
`activator-san`, `activator-ca`, `queue-proxy-ca` etc.

Such deployment and management of the certificates is a big burden for users.

### CODE REVIEW

**Comment by skonto** [comment]
@nak3 pls rebase.

**Comment by nak3** [comment]
@skonto This also needs https://github.com/knative/networking/pull/680.
Could you please review & merge https://github.com/knative/networking/pull/680 first? Otherwise, I need to keep rebasing this PR.

File: pkg/config/config.go [hunk 1]
```diff
@@ -63,6 +63,10 @@ const (
 	// disableHTTP2AnnotationKey is the annotation key attached to a Knative Domain Mapping
 	// to indicate that http2 should not be enabled for it.
 	disableHTTP2AnnotationKey = "kourier.knative.dev/disable-http2"
+
+	// ServingNamespaceEnv is an env variable specifying where the serving is deployed.
+	// e.g. OpenShift deploys Kourier in different namespace so `system.Namespace()` does not work.
```
Comment (skonto) [inline pkg/config/config.go:68]: Afaik this holds for backwards compatibility reasons. Is there anything that stops us from moving everything to `serving` ns?

Comment (nak3) [reply pkg/config/config.go:68]: It is possible but it has a lot of risks to move the gateway into serving ns.

**Comment by nak3** [comment]
@skonto Could you please take a look?

File: pkg/generator/ingress_translator.go [hunk 2]
```diff
@@ -313,7 +310,7 @@ func createUpstreamTLSContext(caCertificate []byte, activatorSAN string, alpnPro
 					},
 					MatchSubjectAltNames: []*envoymatcherv3.StringMatcher{{
 						MatchPattern: &envoymatcherv3.StringMatcher_Exact{
-							Exact: activatorSAN,
+							Exact: certificates.FakeDnsName,
```
Comment (skonto) [inline pkg/generator/ingress_translator.go:313]: I guess if we ever have multiple-knative installations this will have to be changed?

**Pull request merged**